python/ray/dashboard/tests/test_dashboard.py

```diff
@@ -1187,6 +1188,55 @@ def test_dashboard_module_load(tmpdir):
     assert loaded_modules_actual == loaded_modules_expected
 
 
+@pytest.mark.skipif(
+    os.environ.get("RAY_MINIMAL") == "1",
+    reason="This test is not supposed to work for minimal installation.",
+)
+def test_extra_prom_headers_validation(tmpdir):
+    """Test the extra Prometheus headers validation in DashboardHead."""
+    head = DashboardHead(
+        http_host="127.0.0.1",
+        http_port=8265,
+        http_port_retries=1,
+        node_ip_address="127.0.0.1",
+        gcs_address="127.0.0.1:6379",
+        cluster_id_hex=ray.ClusterID.from_random().hex(),
+        grpc_port=0,
+        log_dir=str(tmpdir),
+        temp_dir=str(tmpdir),
+        session_dir=str(tmpdir),
+        minimal=False,
+        serve_frontend=True,
+    )
+    loaded_modules_expected = {"MetricsHead", "DataHead"}
+
+    # Test the base case.
+    head._load_modules(modules_to_load=loaded_modules_expected)
+
+    # Test the supported case.
+    os.environ[PROMETHEUS_HEADERS_ENV_VAR] = '{"H1": "V1", "H2": "V2"}'
+    head._load_modules(modules_to_load=loaded_modules_expected)
+
+    # Test the supported case.
+    os.environ[
+        PROMETHEUS_HEADERS_ENV_VAR
+    ] = '[["H1", "V1"], ["H2", "V2"], ["H2", "V3"]]'
+    head._load_modules(modules_to_load=loaded_modules_expected)
+
+    # Test the unsupported case.
+    with pytest.raises(ValueError):
+        os.environ[PROMETHEUS_HEADERS_ENV_VAR] = '{"H1": "V1", "H2": ["V1", "V2"]}'
```

Comment:
maybe use `monkeypatch.setenv` to avoid env var leak

Reply:
Thanks for the hint. Updated.